lib/SILGen/SILGenProlog.cpp

```diff
@@ -827,11 +828,14 @@ class ArgumentInitHelper {
           }
         }
         break;
-      case ValueOwnership::InOut:
-        assert(isa<MarkMustCheckInst>(value) &&
-               "Expected mark must check inst with inout to be handled in "
-               "emitBBArgs earlier");
+      case ValueOwnership::InOut: {
+        assert(isa<MarkMustCheckInst>(value)
+               && "Expected mark must check inst with inout to be handled in "
```

Comment:
@jckarter I am fine not fixing this in this PR, but as a note in case you get to it before I do, I think it would be fine to actually move the emission of the mark_must_check for inout here so that we do it all in one place.

Eventually, I think all of this code though should be in emitBBArgs.

Reply:
Sounds good, I'll look into doing that on main. It was a bit of a bumpy road just getting to this working state, so I don't want to disrupt 5.9 any more than necessary to get this working, if that's OK with you.